write a single cookie with the correct options

dholdren opened the pull request:
### Problem: httponly setting has no affect
I set `Clearance.configuration.httponly = true` but I noticed that executing `document.cookie` in Chrome's inspector still contained `remember_token`, then I realized that clearance was setting this cookie twice, once with `httponly`, and once without it.

Here's a spec that demonstrates the problem:
https://github.com/CasperSleep/clearance/commit/6f930bc235ed70f7b26750157ef66aaf7d89b30e

```
Failures:

  1) cookie options when httponly config value true sign in should have one remember_token cookie
     Failure/Error: expect(@remember_token_cookies.length).to eq(1), "expected one 'remember_token' cookie:\n#{@remember_token_cookies}"

       expected one 'remember_token' cookie:
       ["remember_token=1b3f09cead4c3ba39bed8ac0831837b0ea52a7b1; path=/; expires=Sat, 05 Jan 2019 18:04:16 -0000; HttpOnly", "remember_token=1b3f09cead4c3ba39bed8ac0831837b0ea52a7b1; path=/"]
     # /Users/dholdren/Dropbox (Personal)/third_party/clearance/spec/requests/cookie_options_spec.rb:41:in `block (4 levels) in <top (required)>'

  2) cookie options when httponly config value true sign in should have the httponly flag set
     Failure/Error: expect(@remember_token_cookies.last).to match(/httponly/i)

       expected "remember_token=fe6e5ed40284f7a724387fd25a2b2524918a9279; path=/" to match /httponly/i
       Diff:
       @@ -1,2 +1,2 @@
       -/httponly/i
       +"remember_token=fe6e5ed40284f7a724387fd25a2b2524918a9279; path=/"

     # /Users/dholdren/Dropbox (Personal)/third_party/clearance/spec/requests/cookie_options_spec.rb:45:in `block (4 levels) in <top (required)>'

  3) cookie options when httponly config value false sign in should have one remember_token cookie
     Failure/Error: expect(@remember_token_cookies.length).to eq(1), "expected one 'remember_token' cookie:\n#{@remember_token_cookies}"

       expected one 'remember_token' cookie:
       ["remember_token=5323f9bdb472031440d20a7283672d6590f856e8; path=/; expires=Sat, 05 Jan 2019 18:04:16 -0000", "remember_token=5323f9bdb472031440d20a7283672d6590f856e8; path=/"]
     # /Users/dholdren/Dropbox (Personal)/third_party/clearance/spec/requests/cookie_options_spec.rb:20:in `block (4 levels) in <top (required)>'

Finished in 0.39193 seconds (files took 1.16 seconds to load)
4 examples, 3 failures

Failed examples:

rspec /Users/dholdren/Dropbox (Personal)/third_party/clearance/spec/requests/cookie_options_spec.rb:40 # cookie options when httponly config value true sign in should have one remember_token cookie
rspec /Users/dholdren/Dropbox (Personal)/third_party/clearance/spec/requests/cookie_options_spec.rb:44 # cookie options when httponly config value true sign in should have the httponly flag set
rspec /Users/dholdren/Dropbox (Personal)/third_party/clearance/spec/requests/cookie_options_spec.rb:19 # cookie options when httponly config value false sign in should have one remember_token cookie
```

### Solution

- removed `add_cookie_to_headers` in Rack middleware
- modified the cookie that is set after login to have the correct options
- reworked all the tests that called `add_cookie_to_headers` explicitly

derekprior commented:
Thanks for this. It’s been a while since I’ve worked in this area of the code so I’ll need to refamiliarize myself. My immediate concern is with removing the cookie from middleware as only setting it on sign in means the expiration won’t be refreshed with each request, correct?

dholdren commented:
good point, I didn't think about that when I did this. I guess there's no existing spec that tests that behaviour?

I'll try to get that working and add a spec.

dholdren commented:
I added a spec (token_expiration_spec.rb) and removed the cookie setting from `Session#sign_in` and restored it in `RackSession#call` so it's still set only once per request, but is set on every request.

dholdren commented:
(a few of the travis jobs need to be restarted)

dholdren commented:
are you able to take another look at this?

dholdren commented:
Can someone restart the build?

rtymchyk commented:
This is surfacing in frustrating ways, such as both cookies not being cleared on logout. Is there any hope for this PR?

dholdren commented:
@mjankowski looks like you're the current maintainer?I've rebased this, fixed the spec issues and resolved all the Hound violations

mjankowski commented:
I'm not THE maintainer but I can take a look at this later, thanks for updating!

dholdren commented:
@mjankowski have you had a chance to review?

mjankowski commented:
Closing after https://github.com/thoughtbot/clearance/pull/804 merge
